### PR TITLE
Sentry integration into play service

### PR DESCRIPTION
**Pull request by gparant**
# Sentry integration
 - Install sentry package
 - Test sentry log
 - Push sentry sourcemaps

### CODE REVIEW

File: back/Dockerfile.sentry
Comment (moufmouf) [inline back/Dockerfile.sentry:1]: I don't understand why you build a separate Dockerfile for this.
The newly pusher image serves no purpose I think.

Instead, you should probably run the "push-sentry-sourcemaps" command in the main Dockerfile, in the build image?

Also, important: the build should keep succeeding if the Sentry secrets are not passed (we are not the only ones running builds on those images)

File: back/src/Enum/EnvironmentVariableValidator.ts [hunk 1]
```diff
@@ -61,6 +61,11 @@ Note that anonymous players don't have any TTL limit because their data is store
             'This email address will be notified if your WorkAdventure version contains a known security flaw. ENABLE_TELEMETRY must be set to "true" for this.'
         ),
     TELEMETRY_URL: z.string().optional().default("https://stats.workadventu.re"),
+    SENTRY_DSN: z.string().optional().describe("If set, WorkAdventure will send errors to Sentry"),
+    SENTRY_RELEASE: z
+        .string()
+        .optional()
+        .describe("If set, WorkAdventure will send errors to Sentry with the good relase version"),
```
Comment (moufmouf) [inline back/src/Enum/EnvironmentVariableValidator.ts:68]:
```suggestion
        .describe("The Sentry release we target. Only used if SENTRY_DSN is configured."),
```

File: back/tsconfig.json [hunk 2]
```diff
@@ -59,14 +59,14 @@
     // "preserveSymlinks": true,              /* Do not resolve the real path of symlinks. */
     // "allowUmdGlobalAccess": true,          /* Allow accessing UMD globals from modules. */
     /* Source Map Options */
-    // "sourceRoot": "",                      /* Specify the location where debugger should locate TypeScript files instead of source locations. */
+    "sourceRoot": "/",                      /* Specify the location where debugger should locate TypeScript files instead of source locations. */
```
Comment (moufmouf) [inline back/tsconfig.json:62]: Are you sure this is necessary? It seems weird but I don't know why you need this.

File: chat/Dockerfile
Comment (moufmouf) [inline chat/Dockerfile:1]: Are the source-maps pushed during the build?

File: map-storage/Dockerfile.sentry
Comment (moufmouf) [inline map-storage/Dockerfile.sentry:1]: Same here, it seems fishy to have a file dedicated to Sentry.

File: map-storage/src/Enum/EnvironmentVariableValidator.ts [hunk 3]
```diff
@@ -33,6 +33,11 @@ const BasicEnvironmentVariables = z.object({
         .describe(
             'The cache-control HTTP header to be used for "normal" ressources. Note: resources containing a hash in the name will be set to "immutable", whatever this setting is.'
         ),
+    SENTRY_DSN: z.string().optional().describe("If set, WorkAdventure will send errors to Sentry"),
+    SENTRY_RELEASE: z
+        .string()
+        .optional()
+        .describe("If set, WorkAdventure will send errors to Sentry with the good relase version"),
```
Comment (moufmouf) [inline map-storage/src/Enum/EnvironmentVariableValidator.ts:40]:
```suggestion
        .describe("The Sentry release we target. Only used if SENTRY_DSN is configured."),
```

File: back/tsconfig.json (hunk 2 above)
Comment (gparant) [reply back/tsconfig.json:62]: We could try without that, but it's a prerequisite in the sentry documentation

File: .env.template [hunk 4]
```diff
@@ -129,5 +129,15 @@ REPORT_ISSUES_URL=
 # LogRocket
 LOGROCKET_ID=
 
+# Sentry integration
+## you be able to set the DSN in the Sentry UI
```
Comment (moufmouf) [inline .env.template:133]:
```suggestion
## Find the DSN in the Sentry UI
```

File: .github/workflows/build-test-and-deploy.yml [hunk 5]
```diff
@@ -786,6 +838,17 @@ jobs:
...
+          SENTRY_DSN_CHAT: ${{ secrets.SENTRY_DSN_CHAT }}
+          SENTRY_DSN_FRONT: ${{ secrets.SENTRY_DSN_FRONT }}
+          SENTRY_DSN_PUSHER: ${{ secrets.SENTRY_DSN_PUSHER }}
+          SENTRY_DSN_MAPSTORAGE: ${{ secrets.SENTRY_DSN_MAPSTORAGE }}
+          SENTRY_DSN_BACK_DEV: ${{ secrets.SENTRY_DSN_BACK_DEV }}
+          SENTRY_DSN_CHAT_DEV: ${{ secrets.SENTRY_DSN_CHAT_DEV }}
+          SENTRY_DSN_FRONT_DEV: ${{ secrets.SENTRY_DSN_FRONT_DEV }}
+          SENTRY_DSN_PUSHER_DEV: ${{ secrets.SENTRY_DSN_PUSHER_DEV }}
+          SENTRY_DSN_MAPSTORAGE_DEV: ${{ secrets.SENTRY_DSN_MAPSTORAGE_DEV }}
+          SENTRY_RELEASE: ${{ github.event_name == 'pull_request' && env.GITHUB_HEAD_REF_SLUG || env.GITHUB_REF_SLUG }}
```
Comment (moufmouf) [inline .github/workflows/build-test-and-deploy.yml:851]: No need for CD envs
```suggestion
```

File: back/Dockerfile [hunk 6]
```diff
@@ -26,11 +26,14 @@ COPY libs ./libs
...
-
+RUN chown -R node:node /usr/src/node_modules
+RUN chown -R node:node /usr/src/package-lock.json
 WORKDIR /usr/src/back
 
 RUN apt-get update && apt-get install -y curl
 HEALTHCHECK --interval=5s --timeout=10s --start-period=5s --retries=10 CMD curl -f http://localhost:8080/ping || exit 1
 
+RUN mkdir ./dist && chown -R node:node ./dist
 USER node
```
Comment (moufmouf) [inline back/Dockerfile:37]: Question: shall we move the "User node" at the top to avoid the "chown" section?

File: deeployer.libsonnet
Comment (moufmouf) [inline deeployer.libsonnet:1]: Remove changes in this file

File: map-storage/Dockerfile [hunk 7]
```diff
@@ -30,8 +30,11 @@ RUN npm ci --omit=dev --workspace map-storage
 COPY libs ./libs
 COPY --from=proto-builder /usr/libs/messages/src ./libs/messages/src
 COPY --chown=node:node map-storage map-storage
-
+RUN chown -R node:node /usr/src/node_modules
+RUN chown -R node:node /usr/src/map-storage/node_modules
+RUN chown -R node:node /usr/src/package-lock.json
 WORKDIR /usr/src/map-storage
+RUN mkdir ./dist && chown -R node:node ./dist
```
Comment (moufmouf) [inline map-storage/Dockerfile:37]: Same. Move USER=node above

**Pull request merged**